src/components/MultiHopTrade/hooks/useTradeExecution/useTradeExecution.ts

```diff
@@ -0,0 +1,115 @@
+import { supportsETH } from '@shapeshiftoss/hdwallet-core'
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+import { useCallback, useState } from 'react'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { usePoll } from 'hooks/usePoll/usePoll'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import type { Swapper2, TradeQuote2 } from 'lib/swapper/api'
+import { SwapperName } from 'lib/swapper/api'
+import { lifi } from 'lib/swapper/swappers/LifiSwapper/LifiSwapper2'
+import { thorchain } from 'lib/swapper/swappers/ThorchainSwapper/ThorchainSwapper2'
+import { assertUnreachable, isEvmChainAdapter } from 'lib/utils'
+import { selectPortfolioAccountMetadataByAccountId } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { TRADE_POLL_INTERVAL_MILLISECONDS } from '../constants'
+import { useAccountIds } from '../useAccountIds'
+import { withFromOrXpub } from './helpers'
+
+export const useTradeExecution = ({
+  swapperName,
+  tradeQuote,
+}: {
+  swapperName: SwapperName
+  tradeQuote: TradeQuote2
+}) => {
+  const [sellTxId, setSellTxId] = useState<string | undefined>()
+  const [buyTxId, setBuyTxId] = useState<string | undefined>()
+  const [message, setMessage] = useState<string | undefined>()
+  const [status, setStatus] = useState<TxStatus>(TxStatus.Unknown)
+  const { poll } = usePoll()
+  const wallet = useWallet().state.wallet
+
+  const { sellAssetAccountId } = useAccountIds({
+    sellAsset: tradeQuote.steps[0].sellAsset,
```

Comment:
Did we want to use `selectFirstHopSellFeeAsset` to keep this in sync with future updates?

Reply:
this one's resolved upstream :)